.envrc

```diff
@@ -1,6 +1,7 @@
 export HAB_DOCKER_OPTS="-p 9636:80 -p 9631:9631 -p 9638:9638"
 export HAB_STUDIO_SUP="--auto-update"
 export HAB_ORIGIN_KEYS
+HAB_ORIGIN=""
```

Comment:
I don't think I understand what benefit is conveyed from adding this line.  😕

Reply:
It prevents the script from failing on line 23:
```bash
if [ -n $HAB_ORIGIN ]; then
  HAB_ORIGIN_KEYS="${HAB_ORIGIN_KEYS},${HAB_ORIGIN}"
fi
```
if `$HAB_ORIGIN` is undefined. This is necessary because of the addition of `set -u`.

Reply:
Ahh, right of course.  I completely missed that, thanks.

Reply:
If this isn't defined we want it to stay that way. Could we switch the `-n` to a `-z`?

Reply:
Whether it's `-n` or `-z` doesn't matter, evaluating:
```bash
if [ -z $HAB_ORIGIN ];
```
Also fails with `HAB_ORIGIN: unbound variable` if the shell sourcing it has `-u` set. The only way to make it safe to run with `-u` is to define variables before they're used. That seems to me like a good practice anyway, no?

Reply:
Defining every variable before it's used feels a little cumbersome to me.

Reply:
FWIW, you can do something like this, too:
```bash
if [ -z "${HAB_ORIGIN:-arbitrary-value-just-so-it-evaluates-to-something}" ]; then
  ...
fi
```

Reply:
The underlying issue here is `HAB_ORIGIN` is an override for the default studio behavior. We leverage that to control some stuff in the VM like key loading. If `HAB_ORIGIN` is unset, the studio will degrade safely. However, if we override it to an empty string or something else the studio will assume our origin is whatever our new default is.

Reply:
I'm not sure I understand your comment, @elliott-davis. `HAB_ORIGIN` is never exported, so whether it's undefined or the empty string shouldn't effect anything outside this particular script, right? And unless I'm mistaken, [the `-n` test](http://tldp.org/LDP/abs/html/refcards.html#AEN22473) should behave the same whether `HAB_ORIGIN` is undefined or the empty string.

As to whether we want to add all of `set -euo pipefail` or not, that seems close to a style issue to me. My concerns on most things style-related is that there's a lot of value in consistency for reducing cognitive load, but I'm loath to spend much time debating the merits of any particular style. I will say that for the purposes of getting these scripts working, it was really helpful to have `-e` and `-u` pointing out where things were failing and since the only code change to accommodate the addition is declaring `HAB_ORIGIN`, I don't see it as cumbersome. Plus, I think there are good reasons to use [`set -euo pipefail` as a default](http://redsymbol.net/articles/unofficial-bash-strict-mode/) for bash scripts to make them easier to debug.

I don't want to get bogged down on this and since I'm new to the project, someone please tell me what they'd approve, and I'll make that change.

Reply:
oh derp. You're right.